### PR TITLE
chore: Add script to re-bind ias e2e services

### PR DESCRIPTION
**Pull request by cschubertcs**
## Context

SAP/cloud-sdk-java-backlog#404.

This PR adds a script to refresh the service bindings of our E2E Apps to the IAS service.
This is necessary to be done once a year, but can be done more often, if wanted.

Once this is merged I will setup a Slack reminder to run this script beginning of 2025.

## Definition of Done

- [x] Functionality scope stated & covered
- [x] ~Tests cover the scope above~
- [x] ~Error handling created / updated & covered by the tests above~
- [x] ~Documentation updated~
- [x] ~Release notes updated~

### CODE REVIEW

**Review by Johannes-Schneider — COMMENTED** [review]
While we are at this: How about introducing a cron GH action to re-new the bindings automatically?

Also: Are you sure the new certificates are valid for one **year**? Thought it was just one **month**.

**Comment by cschubertcs** [comment]
> While we are at this: How about introducing a cron GH action to re-new the bindings automatically?

Yeah, I was thinking the same, but wasn't sure whether it makes sense to prepare a script that will be run in a year, but might be outdated by then. With "outdated" I mean that we might have fewer or more bindings to be updated, different structure or what not. So I was a bit hesitant. However, we can adjust the validity to do it maybe every two month. That way we also verify, that the script/process still works. Any thoughts?
 
> Also: Are you sure the new certificates are valid for one **year**? Thought it was just one **month**.

I found out that it is configurable. We can set basically any validity, as I couldn't find any maximum in the documentation.

File: .pipeline/scripts/refresh-ias-certificates.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    Binding("scp-cf-spring-ias-test-approuter", "cf-spring-ias", "ias-router"),
+]
+
+for binding in bindings:
+    # Information on the parameter can be found here:
+    # https://github.wdf.sap.corp/CPSecurity/Knowledge-Base/blob/master/08_Tutorials/iasbroker/README.md#service-binding-creation
+    binding_parameter = (f'{{"credential-type": "X509_GENERATED", "app-identifier": "{binding.app_identifier}", '
+                         f'"validity": 1, "validity-type": "YEARS"}}')
+    subprocess.run(["cf", "unbind-service", binding.app_name, binding.service_name])
+    subprocess.run(["cf", "bind-service", binding.app_name, binding.service_name, "-c", binding_parameter])
```
[the remark this answers is not in the document]
Comment (cschubertcs) [reply .pipeline/scripts/refresh-ias-certificates.py:33]: Yes, I ran it and it updated all bindings to the 1 year validity.

**Review by MatKuhr — COMMENTED** [review]
I am wondering why we have this here and not on our E2E test repo? Might be more natural to look there, and the E2E test instance names we reference here are also maintained in that repo...

Also, I see the K8S app isn't listed, but I guess that still uses old-school client secret and doesn't need rotation 😄

**Comment by cschubertcs** [comment]
> I am wondering why we have this here and not on our E2E test repo? Might be more natural to look there, and the E2E test instance names we reference here are also maintained in that repo...

Hm, true, very good question. I had the intention of building some automation around it. For that I was thinking of github actions, so it made sense to have it here. But splitting this from the E2E tests, which is definitely belongs to, doesn't make so much sense.
Will move it over and build a small Jenkins Pipeline around it.

> Also, I see the K8S app isn't listed, but I guess that still uses old-school client secret and doesn't need rotation 😄

I took the list of applications based on what failed last week with the invalid credentials. So yes, I would assume it's not using certificated based IAS integration.

**Comment by cschubertcs** [comment]
Any thoughts on the validity of the certificates and the frequency of running the script? My feeling is that once a year is to low frequency to be sure it's actually working. On the other hand the least amount of problems per timeframe would be to set the validity very high (multiple years, if possible) and just "forget" about this completely.

I'm torn and would want a team opinion here.

**Comment by Johannes-Schneider** [comment]
@cschubertcs wrote:

> For that I was thinking of github actions, so it made sense to have it here. But splitting this from the E2E tests, which is definitely belongs to, doesn't make so much sense.
Will move it over and build a small Jenkins Pipeline around it.

AFAIK, we **could** also use GH actions in our E2E test repo - at least the github instance is capable of doing that and I can see that there is a toggle within our repo settings to enable actions.
But I guess we should clarify how this would work in terms of costs etc.

---

> Any thoughts on the validity of the certificates and the frequency of running the script? My feeling is that once a year is to low frequency to be sure it's actually working.

Personally, I feel like choosing a rather low frequency should be ok.
We could run the pipeline manually once to verify that everything is indeed working.
And we could implement some notification mechanism (i.e. Slack post) in case anything unexpectedly fails during the pipeline run.

**Comment by cschubertcs** [comment]
As discussed here and in Teams I will move this script over to the E2E-Test repo. Also, as the future here is not clear, I will just setup a reminder to run this script instead of spending effort for an action/pipeline.

**Pull request closed without merging**